src/router_core/modules/mobile_sync/mobile.c

```diff
@@ -191,15 +190,66 @@ static qd_composed_field_t *qcm_mobile_sync_message_headers(const char *address,
 }
 
 
+/**
+ * An address descriptor is a list with positional identification of values.
+ *   Position    Value
+ *   ================================
+ *    0          Address hash key
+ *    1          Treatment value
+ *    2          Local in-link count
+ *    3          Local out-link capacity
+ */
+static void qcm_mobile_sync_compose_addr_descriptor(const qdr_address_t *addr, qd_composed_field_t *field, bool is_added)
+{
+    const char *hash_key = (const char*) qd_hash_key_by_handle(addr->hash_handle);
+
+    qd_compose_start_list(field);
+    qd_compose_insert_string(field, hash_key);
+    if (is_added) {
+        qd_compose_insert_int(field, addr->treatment);
+        qd_compose_insert_int(field, DEQ_SIZE(addr->inlinks));
+    }
+    qd_compose_end_list(field);
+}
+
+
+static qd_iterator_t *qcm_mobile_sync_parse_addr_descriptor(qd_parsed_field_t *field, int *treatment, int *inlink_count)
+{
+    qd_iterator_t *iter = 0;
```

Comment:
As a first line of code, can we please check for a non-zero field like we do here - https://github.com/apache/qpid-dispatch/blob/main/src/parse.c#L359 or like we do in the qcm_mobile_sync_router_by_id() function

Reply:
so, `assert` then?